straightline.py

```diff
@@ -105,26 +119,81 @@ def process_image(self, image, screen):
                     self.drive.move(turn, 0)
                 self.last_t_error = self.t_error
             else:
-                self.stop_and_wait()
+                if self.driving:
+                    self.stop_and_wait()
         else:
-            # No markers found
-            self.stop_and_wait()
+            logger.info("no marker, looking for ribbon")
+            self.m_found = False
+            cropped_image = cv2.pyrDown(image, dstsize=(int(self.image_centre_x), int(self.image_centre_y)))
+            cropped_image = cropped_image[self.LINE_CROP_BOTTOM:self.LINE_CROP_TOP, self.LINE_CROP_LEFT:self.LINE_CROP_RIGHT]
+            img = cv2.cvtColor(cropped_image, cv2.COLOR_BGR2RGB)
+            ribbon_frame = pygame.surfarray.make_surface(cv2.flip(img, 1))
+            screen.blit(ribbon_frame, (0, 0))
+            blur_image = cv2.medianBlur(cropped_image, 3)
+            blur_image = cv2.cvtColor(blur_image, cv2.COLOR_RGB2HSV)
+            ribbon_mask = threshold_image(blur_image, self.ribbon_colour)
+            self.ribbon_following(ribbon_mask)
+            
         # Display the resulting frame
         frame = pygame.surfarray.make_surface(cv2.flip(frame,1))
-        screen.fill([0,0,0])
-        screen.blit(frame, (0,0))
+        screen.blit(frame, (self.LINE_CROP_TOP,0))
         pygame.display.update()
-        found_identifier = "F" if self.found else "NF"
-        img_name = "%d%simg.jpg" % (self.i, found_identifier)
+        m_found_identifier = "mF" if self.m_found else "NmF"
+        l_found_identifier = "lF" if self.l_found else "NlF"
+        img_name = "%d%s%s%dimg.jpg" % (self.i, m_found_identifier, l_found_identifier, self.ribbon_pos)
         # filesave for debugging: 
-        # cv2.imwrite(img_name, gray)
+#        if self.driving:
```

Comment:
minor: remove unused code?

Reply:
I should make a command line parameter to enable image saving, as this comes up in every challenge.